src/utils/MainLogger.php

```diff
@@ -153,6 +153,37 @@ public function setLogDebug(bool $logDebug) : void{
 	 * @param array|null $trace
 	 */
 	public function logException(\Throwable $e, $trace = null){
+		$messages = [];
+		self::logExceptionToArray($e, $trace, 0, $messages);
+
+		$this->synchronized(function() use ($messages) : void{
+			foreach($messages as $depth => [$message, $stack]) {
+				if($depth === 0){
+					$this->critical($message);
+				}else{
+					$this->debug(str_repeat("  ", $depth - 1) . "Caused by:", true);
+					$this->debug(str_repeat("  ", $depth) . $message, true);
+				}
+				foreach($stack as $line){
+					$this->debug($line, true);
+				}
+			}
+		});
+
+		$this->syncFlushBuffer();
+	}
+
+	/**
+	 * Appends exceptions messages recursively to the referenced array.
+	 *
+	 * Each element of the array is a [string, string[]], containing the message and the trace lines.
+	 *
+	 * @param \Throwable                $e
+	 * @param array|null                $trace
+	 * @param int                       $depth
+	 * @param array<string|string[]> &$ret
```

Comment:
This doc comment is actually wrong. No idea why phpstan didn't pick it up.

Help needed with specifying tuples in phpdoc.

Reply:
phpstan doesn't like reference notation, it might need to be removed